e2e/testcases/oci_sync_test.go

```diff
@@ -397,6 +398,8 @@ func TestSwitchFromGitToOci(t *testing.T) {
 		Image: imageURL,
 		Auth:  configsync.AuthNone,
 	}
+	// Enable deletion propagation for RepoSync
+	nomostest.EnableDeletionPropagation(repoSyncOCI)
```

Comment:
RepoSyncObjectV1Beta1 should do this for you

Reply:
Not the `fake.RepoSyncObjectV1Beta1` that is currently used. We would need to switch to `nomostest.RepoSyncObjectV1Beta1` and reconfigure the spec from `git` -> `oci`. I don't mind this refactor, although seems like a bit of a wash with the extra setup required

Reply:
Maybe we need a nomostest.RepoSyncObjectV1Beta1WithOCI ?

I'd prefer the tests get the annotation implicitly, without needing to know about it. But this is obviously a bit of an edge case where the cleanup needs to know how it works tho.

Reply:
Refactored, PTAL